include/flashinfer/pos_enc.cuh

```diff
@@ -746,6 +747,217 @@ __global__ void BatchQKApplyRotaryKernel(
     FLASHINFER_ERROR("Unsupported rope_dim. Supported values: 16, 32, 64, 128, 256"); \
   }
 
+/*!
+ * \brief Unified CUDA kernel to apply RoPE, quantize to FP8, and append to paged cache.
+ *
+ * Templated on CacheT to support both GQA/MHA (paged_kv_t) and MLA (paged_kv_mla_t).
+ * Cache-only behaviors are selected with constexpr on the CacheT.
+ */
+template <bool interleave, uint32_t vec_size, uint32_t bdx, typename DType, typename IdType,
+          typename QuantType, bool IS_MLA, typename CacheT>
```

Comment:
Are `IS_MLA` and `CacheT` duplicated? Isn't `IS_MLA` just `std::is_same<CacheT, paged_kv_mla_t>::value`?